app/interactors/album.js

```diff
@@ -0,0 +1,25 @@
+const Album = require('../models').albums;
+
+exports.purchaseAlbum = (user, album) => {
+
+  return new Promise((resolve, reject) => {
+
+    Album.findOne({ where: {id: album.id, userId : user.id} }).then(alreadyPurchased => {
+
+      if(alreadyPurchased){
+        resolve(false);
+      }else{
+        Album.create({
+          id: album.id,
+          userId: user.id,
+          title: album.title
+        }).then(success => {
+          resolve(success);
```

Comment:
To keep consistency it would be nice here to resolve into a true, so this functions always returns a boolean.